components/AccountBalance.js

```diff
@@ -2,37 +2,90 @@ import { useEffect, useState } from "react";
 import { formatNumber } from "@/lib/utils";
 import styled from "styled-components";
 
-export default function AccountBalance({ transactions }) {
+export default function AccountBalance({
+  transactions,
+  filteredAccount,
+  filteredAccountType,
+}) {
   const [currentBalance, setCurrentBalance] = useState(0);
 
   useEffect(() => {
-    const sumOfTransaction = transactions.reduce((total, transaction) => {
-      return transaction.type === "income"
-        ? total + transaction.amount
-        : total - transaction.amount;
-    }, 0);
-    setCurrentBalance(sumOfTransaction);
+    const incomeTotal = transactions
+      .filter((transaction) => transaction.type === "income")
+      .reduce((total, transaction) => total + transaction.amount, 0);
+
+    const expenseTotal = transactions
+      .filter((transaction) => transaction.type === "expense")
+      .reduce((total, transaction) => total + transaction.amount, 0);
+
+    setCurrentBalance(incomeTotal - expenseTotal);
   }, [transactions]);
```

Comment:
this useEffect is not needed. you can calculate incomeTotal and expenseTotal in the component directly.